DEVELOPING.md

```diff
@@ -1,210 +1,205 @@
-* `hab studio enter`
-* `start-builder`
+You will need to create an OAuth application in GitHub, and use the private key, client id and client secret from the app to configure Builder's environment (below).
+
+`APP_HOSTNAME` mentioned below, will typically be `localhost`.
+
+However, if you are going to be doing Web UI development, and running the Web UI on your Host OS, then you will need to use `localhost:3000` instead of `localhost` for `APP_HOSTNAME`.
 
-### Merging the shards
+1. [Create a new GitHub application](https://github.com/settings/apps/new) in your GitHub account
+1. Give it a meaningful `GitHub App name`, e.g., "Builder Local Dev"
+1. Set the value of `Homepage URL` to `http://${APP_HOSTNAME}`
+1. Set the value of `User authorization callback URL` to `http://${APP_HOSTNAME}/` (The trailing `/` is *important*)
+1. Set the value of `Webhook URL` to `http://${APP_HOSTNAME}/`
+1. Set Repository metadata, Repository administration, Repository content and Organization members to read only (this is only used for your org so it's safe)
+1. Download and save the private key. It will result in a file like `app-name.date.private-key.pem`
+1. Record the the client-id, client-secret, app_id and public page link (in the left sidebar). These will be used for the `GITHUB_CLIENT_ID`, `GITHUB_CLIENT_SECRET`, `GITHUB_APP_ID` and `GITHUB_APP_URL` config variables in the section below.
 
-If you created your development environment before August 17th 2018, you have
-a database that contains 128 shards. You need to merge them into one, or your
-services will flap and spew errors into the log that say "Shard migration
-hasn't been completed successfully".
+### Builder configuration
 
-To migrate your shards, run the following commands from inside your development
-studio:
+1. Copy the GitHub application private key (from section above) to the following location (_Important: name it exactly as shown_) `${BUILDER_SRC_ROOT}/.secrets/builder-github-app.pem`
+1. Make a copy of the sample env file: `cp ${BUILDER_SRC_ROOT}/.secrets/habitat-env.sample ${BUILDER_SRC_ROOT}/.secrets/habitat-env`
+1. Edit the env file with your favorite editor `${BUILDER_SRC_ROOT}/.secrets/habitat-env` and populate the variables appropriately
+1. Save and close the env file
 
+## Builder Services Setup
 
-```
-PGPASSWORD=$(cat /hab/svc/builder-datastore/config/pwfile) tools/merge-shards/merge-shards.sh jobsrv migrate
-PGPASSWORD=$(cat /hab/svc/builder-datastore/config/pwfile) tools/merge-shards/merge-shards.sh sessionsrv migrate
-PGPASSWORD=$(cat /hab/svc/builder-datastore/config/pwfile) tools/merge-shards/merge-shards.sh originsrv migrate
-```
+### Starting Builder services
+Once the Builder Repo is configured, Builder services can be started inside the Habitat Studio.
 
-If all goes well, you'll see some messages about things being all good, and
-your services should behave normally. If they don't, exit the studio and
-re-enter.
+* `cd ${BUILDER_SRC_ROOT}`
+* `direnv allow`
```

Comment:
Is direnv installed by default? Maybe add `apt-get install direnv`

Reply:
direenv is installed by the provision.sh script so we should be ok